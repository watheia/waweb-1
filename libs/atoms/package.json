{
  "name": "@waweb/atoms",
  "version": "0.0.1"
}
