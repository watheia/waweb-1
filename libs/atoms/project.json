{
  "root": "libs/atoms",
  "sourceRoot": "libs/atoms/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/atoms",
        "tsConfig": "libs/atoms/tsconfig.lib.json",
        "project": "libs/atoms/package.json",
        "entryFile": "libs/atoms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/atoms/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/atoms/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/atoms"],
      "options": {
        "jestConfig": "libs/atoms/jest.config.js",
        "passWithNoTests": true
      }
    },
    "storybook": {
      "executor": "@nrwl/storybook:storybook",
      "options": {
        "uiFramework": "@storybook/react",
        "port": 4400,
        "config": {
          "configFolder": "libs/atoms/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "build-storybook": {
      "executor": "@nrwl/storybook:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "uiFramework": "@storybook/react",
        "outputPath": "dist/storybook/ui",
        "config": {
          "configFolder": "libs/atoms/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    }
  }
}
