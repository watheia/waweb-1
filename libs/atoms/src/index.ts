export { default as styleUtils } from './lib/utils.module.css';
export { default as Container } from './lib/container';
export { default as LoadingDots } from './lib/loading-dots';
export { default as Logo } from './lib/logo';

export type { HeaderProps, HeaderElement, HeaderVariant } from './lib/header';
export { default as Header } from './lib/header';

export { default as MessageInput } from './lib/message-input';
export { default as NProgress } from './lib/nprogress';
export { default as Page } from './lib/page';
export { default as PageSpinner } from './lib/page-spinner';
export { default as ResizeHandler } from './lib/resize-handler';
export { default as Select } from './lib/select';

export type { TextProps, TextElement, TextVariant } from './lib/text';
export { default as Text } from './lib/text';

export { default as ViewSource } from './lib/view-source';
