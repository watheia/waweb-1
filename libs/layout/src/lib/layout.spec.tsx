import { render } from '@testing-library/react';

import Layout from './layout';
import { MockContent } from '@waweb/atoms';

describe('waweb.layout', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Layout>
        <MockContent />
      </Layout>
    );
    expect(container).toBeInstanceOf(HTMLDivElement);
    expect(container).toMatchSnapshot();
  });
});
