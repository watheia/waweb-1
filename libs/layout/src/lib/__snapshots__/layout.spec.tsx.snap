// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`waweb.layout should render successfully 1`] = `
<div>
  <svg
    class="svg"
    fill="#151513"
    height="64"
    viewBox="0 0 250 250"
    width="64"
    xmlns="http://www.w3.org/2000/svg"
  >
    <a
      href="https://gitlab.com/watheia/waweb"
      rel="noopener noreferrer"
      target="_blank"
      title="View Source"
    >
      <path
        d="M0 0l115 115h15l12 27 108 108V0z"
        fill="#fff"
      />
      <path
        class="arm"
        d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
      />
      <path
        d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
      />
    </a>
  </svg>
  <div
    class=""
  >
    <header
      class="header"
    >
      <div
        class="header-logos"
      >
        <button
          aria-label="Mobile Menu"
          class="button"
          type="button"
        >
          <svg
            class="icon"
            fill="none"
            height="24"
            shape-rendering="geometricPrecision"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M3 12h18"
            />
            <path
              d="M3 6h18"
            />
            <path
              d="M3 18h18"
            />
          </svg>
        </button>
        <a
          class="logo"
          href="/"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="tabs"
      >
        <a
          class="tab"
          href="/home"
        >
          Home
        </a>
        <a
          class="tab"
          href="/blog"
        >
          Blog
        </a>
        <a
          class="tab"
          href="/support"
        >
          Support
        </a>
        <a
          class="tab"
          href="/expo"
        >
          Expo
        </a>
      </div>
    </header>
    <div
      class="page"
    >
      <div
        class="app-messages absolute top-6 z-10 flex w-screen flex-col place-items-center justify-center"
      />
      <div
        data-reach-skip-nav-content=""
        id="reach-skip-nav"
      />
      <main
        class="main"
      >
        <div
          class="full"
        >
          <div
            class="relative overflow-hidden bg-white py-16"
          >
            <div
              class="hidden lg:absolute lg:inset-y-0 lg:block lg:h-full lg:w-full"
            >
              <div
                aria-hidden="true"
                class="relative mx-auto h-full max-w-prose text-lg"
              >
                <svg
                  class="absolute top-12 left-full translate-x-32 transform"
                  fill="none"
                  height="384"
                  viewBox="0 0 404 384"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="74b3fd99-0a6f-4271-bef2-e80eeafdf357"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-gray-200"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#74b3fd99-0a6f-4271-bef2-e80eeafdf357)"
                    height="384"
                    width="404"
                  />
                </svg>
                <svg
                  class="absolute top-1/2 right-full -translate-y-1/2 -translate-x-32 transform"
                  fill="none"
                  height="384"
                  viewBox="0 0 404 384"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="f210dbf6-a58d-4871-961e-36d5016a0f49"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-gray-200"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#f210dbf6-a58d-4871-961e-36d5016a0f49)"
                    height="384"
                    width="404"
                  />
                </svg>
                <svg
                  class="absolute bottom-12 left-full translate-x-32 transform"
                  fill="none"
                  height="384"
                  viewBox="0 0 404 384"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="d3eb07ae-5182-43e6-857d-35c643af9034"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-gray-200"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#d3eb07ae-5182-43e6-857d-35c643af9034)"
                    height="384"
                    width="404"
                  />
                </svg>
              </div>
            </div>
            <div
              class="relative px-4 sm:px-6 lg:px-8"
            >
              <div
                class="mx-auto max-w-prose text-lg"
              >
                <h1>
                  <span
                    class="block text-center text-base font-semibold uppercase tracking-wide text-teal-600"
                  >
                    Introducing
                  </span>
                  <span
                    class="mt-2 block text-center text-3xl font-extrabold leading-8 tracking-tight text-gray-900 sm:text-4xl"
                  >
                    JavaScript for Beginners
                  </span>
                </h1>
                <p
                  class="mt-8 text-xl leading-8 text-gray-500"
                >
                  Aliquet nec orci mattis amet quisque ullamcorper neque, nibh sem. At arcu, sit dui mi, nibh dui, diam eget aliquam. Quisque id at vitae feugiat egestas ac. Diam nulla orci at in viverra scelerisque eget. Eleifend egestas fringilla sapien.
                </p>
              </div>
              <div
                class="prose prose-teal prose-lg mx-auto mt-6 text-gray-500"
              >
                <p>
                  Faucibus commodo massa rhoncus, volutpat. 
                  <strong>
                    Dignissim
                  </strong>
                   
                  sed 
                  <strong>
                    eget risus enim
                  </strong>
                  . Mattis mauris semper sed amet vitae sed turpis id. Id dolor praesent donec est. Odio penatibus risus viverra tellus varius sit neque erat velit. Faucibus commodo massa rhoncus, volutpat. Dignissim sed eget risus enim.
                   
                  <a
                    href="#"
                  >
                    Mattis mauris semper
                  </a>
                   sed amet vitae sed turpis id.
                </p>
                <ul
                  role="list"
                >
                  <li>
                    Quis elit egestas venenatis mattis dignissim.
                  </li>
                  <li>
                    Cras cras lobortis vitae vivamus ultricies facilisis tempus.
                  </li>
                  <li>
                    Orci in sit morbi dignissim metus diam arcu pretium.
                  </li>
                </ul>
                <p>
                  Quis semper vulputate aliquam venenatis egestas sagittis quisque orci. Donec commodo sit viverra aliquam porttitor ultrices gravida eu. Tincidunt leo, elementum mattis elementum ut nisl, justo, amet, mattis. Nunc purus, diam commodo tincidunt turpis. Amet, duis sed elit interdum dignissim.
                </p>
                <h2>
                  From beginner to expert in 30 days
                </h2>
                <p>
                  Id orci tellus laoreet id ac. Dolor, aenean leo, ac etiam consequat in. Convallis arcu ipsum urna nibh. Pharetra, euismod vitae interdum mauris enim, consequat vulputate nibh. Maecenas pellentesque id sed tellus mauris, ultrices mauris. Tincidunt enim cursus ridiculus mi. Pellentesque nam sed nullam sed diam turpis ipsum eu a sed convallis diam.
                </p>
                <blockquote>
                  <p>
                    Sagittis scelerisque nulla cursus in enim consectetur quam. Dictum urna sed consectetur neque tristique pellentesque. Blandit amet, sed aenean erat arcu morbi.
                  </p>
                </blockquote>
                <p>
                  Faucibus commodo massa rhoncus, volutpat. Dignissim sed eget risus enim. Mattis mauris semper sed amet vitae sed turpis id. Id dolor praesent donec est. Odio penatibus risus viverra tellus varius sit neque erat velit.
                </p>
                <figure>
                  <img
                    alt=""
                    class="w-full rounded-lg"
                    height="873"
                    src="https://images.unsplash.com/photo-1500648767791-00dcc994a43e?ixlib=rb-1.2.1&auto=format&fit=facearea&w=1310&h=873&q=80&facepad=3"
                    width="1310"
                  />
                  <figcaption>
                    Sagittis scelerisque nulla cursus in enim consectetur quam.
                  </figcaption>
                </figure>
                <h2>
                  Everything you need to get up and running
                </h2>
                <p>
                  Purus morbi dignissim senectus mattis 
                  <a
                    href="#"
                  >
                    adipiscing
                  </a>
                  . Amet, massa quam varius orci dapibus volutpat cras. In amet eu ridiculus leo sodales cursus tristique. Tincidunt sed tempus ut viverra ridiculus non molestie. Gravida quis fringilla amet eget dui tempor dignissim. Facilisis auctor venenatis varius nunc, congue erat ac. Cras fermentum convallis quam.
                </p>
                <p>
                  Faucibus commodo massa rhoncus, volutpat. Dignissim sed eget risus enim. Mattis mauris semper sed amet vitae sed turpis id. Id dolor praesent donec est. Odio penatibus risus viverra tellus varius sit neque erat velit.
                </p>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="footer border-t border-dotted border-gray-700 bg-gray-900/75"
        data-testid="waweb.layout/footer"
      >
        <div
          class="footer-legal"
        >
          <div
            class="footer-copyright"
            data-testid="waweb.layout/footer:copyright"
          >
            Copyright © 
            2022 
             Watheia Labs, LLC. All rights reserved.
          </div>
          <div
            class="footer-center-group"
          >
            <p
              class="footer-paragraph"
            >
              <a
                class="footer-link"
                data-testid="waweb.layout/footer:view-source"
                href="https://gitlab.com/watheia/waweb"
                rel="noopener noreferrer"
                target="_blank"
              >
                Source Code
              </a>
            </p>
            <div
              class="footer-separator"
            />
            <p
              class="footer-paragraph"
            >
              <a
                class="footer-link"
                data-testid="waweb.layout/footer:legal"
                href="#"
                rel="noopener noreferrer"
                target="_blank"
              >
                Legal
              </a>
            </p>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
